### DISCUSSION TITLE
How to feed command line argument to git-bash.exe?

### DISCUSSION THREAD

**Discussion by diffuseinstability** [git-for-windows/git#4813] [Q&A] [answered]
I'm running Stable Diffusion on Windows 11 with the Automatic1111 Web UI using the Git Bash (git-bash.exe) emulation layer (https://www.atlassian.com/git/tutorials/git-bash). Currently, I have to launch git-bash.exe using a shortcut I created in my Start Menu, copy and paste in a long command to change the current directory, then copy and paste another long command to run webui-user.bat. I can't even use hotkeys, because Ctrl+V doesn't work in Git Bash.

What I want to be able to do is create a shortcut or .bat file that **_feeds Git Bash a command line argument_** instructing it to run a (different) .bat file that, in turn, contains the commands to change the directory and run webui-user.bat. In other words, I want a a shortcut or .bat file will use the Windows OS to run Git Bash, but also feed Git Bash a command to run with the emulation environment.

**Does anyone know how to feed such a command line argument to git-bash.exe?**

Failing that, does anyone know where to any other method for setting a 1-click execution of this setup? Or even a list of commands that can be executed on the command line for git-bash.exe?

For arguments sake, let's assume 

Git Bash executable: C:\Program Files\Git\git-bash.exe

Automatic1111 UI: C:\Users\thisuser\SD\webui-user.bat

Default directory used as Git Bash "home" upon start up: C:\Users\thisuser\

**Comment by dscho** [comment]
Do you need Git Bash at all? If all you want to run a `.bat`, then just run that `.bat`, using CMD. If you chose the option to add Git to the `PATH` (the default, which appends `;C:\Program Files\Git\cmd` to the environment variable `PATH`), that Batch file should have access to `git.exe` and it should not need to involve Bash at all.

**Reply by diffuseinstability** [reply]
I'm not trying to run Git. I'm trying to run Stable Diffusion using Git Bash.

**Reply by dscho** [reply, marked as answer]
But why the need for Bash? If all you want is to run a Batch file?
